app/javascript/shared/components/ui/LabelDropdown.vue

```diff
@@ -0,0 +1,212 @@
+<template>
+  <div class="dropdown-search-wrap">
+    <h4 class="text-block-title">
+      {{ $t('CONTACT_PANEL.LABELS.LABEL_SELECT.TITLE') }}
+    </h4>
+    <div class="search-wrap">
+      <input
+        ref="searchbar"
+        v-model="search"
+        type="text"
+        class="search-input"
+        autofocus="true"
+        placeholder="Filter"
+      />
+    </div>
+    <div class="list-wrap">
+      <div class="list">
+        <woot-dropdown-menu>
+          <woot-dropdown-item
+            v-for="label in filteredActiveLabels"
+            :key="label.title"
+          >
+            <div class="name-label-icon-wrap">
+              <button class="button clear" @click="() => onAddRemove(label)">
+                <div class="name-label-wrap">
+                  <div
+                    v-if="label.color"
+                    class="label-color--display"
+                    :style="{ backgroundColor: label.color }"
+                  />
+                  <span>{{ label.title }}</span>
+                </div>
+                <i
+                  v-if="selectedLabels.includes(label.title)"
+                  class="icon ion-checkmark-round"
+                />
+              </button>
+            </div>
+          </woot-dropdown-item>
+        </woot-dropdown-menu>
+        <div v-if="noResult" class="no-result">
+          {{ $t('CONTACT_PANEL.LABELS.LABEL_SELECT.NO_RESULT') }}
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import WootDropdownItem from 'shared/components/ui/dropdown/DropdownItem.vue';
+import WootDropdownMenu from 'shared/components/ui/dropdown/DropdownMenu.vue';
+export default {
+  components: {
+    WootDropdownItem,
+    WootDropdownMenu,
+  },
+
+  props: {
+    conversationId: {
+      type: [String, Number],
+      required: true,
+    },
+    accountLabels: {
+      type: Array,
+      default: () => [],
+    },
+    selectedLabels: {
+      type: Array,
+      default: () => [],
+    },
+    updateLabels: {
+      type: Function,
+      default: () => {},
+    },
+    onAdd: {
+      type: Function,
+      default: () => {},
+    },
+    onRemove: {
+      type: Function,
+      default: () => {},
```

Comment:
The best here is to use emit events from the component, instead of passing down the method.